apps/etterlatte-behandling/src/main/kotlin/grunnlagsendring/doedshendelse/DoedshendelseService.kt

```diff
@@ -95,51 +97,78 @@ class DoedshendelseService(
-        nyeBeroerte.forEach { barn ->
+        val nyeBeroerte =
+            beroerte.map { PersonFnrMedRelasjon(it.person.foedselsnummer.value, it.relasjon) }
+                .filter { !eksisterendeBeroerte.contains(it.fnr) }
+        nyeBeroerte.forEach { person ->
             doedshendelseDao.opprettDoedshendelse(
                 DoedshendelseInternal.nyHendelse(
                     avdoedFnr = avdoed.foedselsnummer.verdi.value,
                     avdoedDoedsdato = avdoed.doedsdato!!.verdi,
-                    beroertFnr = barn,
-                    relasjon = Relasjon.BARN,
+                    beroertFnr = person.fnr,
+                    relasjon = person.relasjon,
                     endringstype = endringstype,
                 ),
             )
         }
     }
 
+    private fun oppdaterDodshendelser(doedshendelserForAvdoed: List<DoedshendelseInternal>) {
+        doedshendelserForAvdoed.forEach {
+            doedshendelseDao.oppdaterDoedshendelse(it)
+        }
+    }
+
     private fun lagreDoedshendelser(
-        beroerte: List<Person>,
+        beroerte: List<PersonMedRelasjon>,
         avdoed: PersonDTO,
         endringstype: Endringstype,
     ) {
-        beroerte.forEach { barn ->
+        beroerte.forEach { person ->
             doedshendelseDao.opprettDoedshendelse(
                 DoedshendelseInternal.nyHendelse(
                     avdoedFnr = avdoed.foedselsnummer.verdi.value,
                     avdoedDoedsdato = avdoed.doedsdato!!.verdi,
-                    beroertFnr = barn.foedselsnummer.value,
-                    relasjon = Relasjon.BARN,
+                    beroertFnr = person.person.foedselsnummer.value,
+                    relasjon = person.relasjon,
                     endringstype = endringstype,
                 ),
             )
         }
     }
 
-    private fun finnBeroerteBarn(avdoed: PersonDTO): List<Person> {
+    private fun finnBeroerteBarn(avdoed: PersonDTO): List<PersonMedRelasjon> {
         val maanedenEtterDoedsfall = avdoed.doedsdato!!.verdi.plusMonths(1).withDayOfMonth(1)
 
         return with(avdoed.avdoedesBarn ?: emptyList()) {
             this.filter { barn -> barn.doedsdato == null }
                 .filter { barn -> barn.under20PaaDato(maanedenEtterDoedsfall) }
+                .map { PersonMedRelasjon(it, Relasjon.BARN) }
         }
     }
+
+    private fun finnBeroerteEpser(avdoed: PersonDTO): List<PersonMedRelasjon> {
+        return avdoed.sivilstand?.filter { it.verdi.relatertVedSiviltilstand?.value !== null }
+            ?.filter {
+                it.verdi.sivilstatus == Sivilstatus.UGIFT ||
+                    it.verdi.sivilstatus == Sivilstatus.UGIFT ||
+                    it.verdi.sivilstatus == Sivilstatus.SKILT ||
+                    it.verdi.sivilstatus == Sivilstatus.ENKE_ELLER_ENKEMANN ||
+                    it.verdi.sivilstatus == Sivilstatus.SEPARERT
+            }
+            ?.map {
+                pdlTjenesterKlient.hentPdlModell(
```

Comment:
Denne føles litt unødvendig ut kun for å få riktig type med videre (bruker ikke dataen) 🤔 Ser jo at vi må ha en Person-type som vi har benyttet for barna, men det brukes ikke til noe spesielt, så kunne kanskje skrevet om til å bare bruke `PersonFnrMedRelasjon`?

Reply:
Godt poeng!